packages/math-input/src/components/keypad/button.tsx

```diff
@@ -44,60 +49,76 @@ const styles = StyleSheet.create({
+        const {children, style, tintColor, gestureManager, keyConfig} =
+            this.props;
+        const hovered = false;
+        const focused = false;
+        const pressed = false;
         return (
-            <View style={style}>
-                <Clickable onClick={onPress} style={styles.clickable}>
-                    {({hovered, focused, pressed}) => {
-                        return (
-                            <View
-                                style={[
-                                    styles.outerBoxBase,
-                                    hovered && styles.outerBoxHover,
-                                    pressed && styles.outerBoxPressed,
-                                ]}
-                            >
-                                <View
-                                    style={[
-                                        styles.base,
-                                        tintColor != null
-                                            ? {background: tintColor}
-                                            : undefined,
-                                        hovered && styles.hovered,
-                                        focused && styles.focused,
-                                        pressed && styles.pressed,
-                                    ]}
-                                >
-                                    {children}
-                                </View>
-                            </View>
-                        );
-                    }}
-                </Clickable>
+            <View
+                style={style}
+                onTouchStart={(evt) =>
+                    gestureManager.onTouchStart(evt, keyConfig.id)
+                }
+                onTouchEnd={(evt) => gestureManager.onTouchEnd(evt)}
+                onTouchMove={(evt) => gestureManager.onTouchMove(evt)}
+                onTouchCancel={(evt) => gestureManager.onTouchCancel(evt)}
+            >
+                <View
+                    style={[
+                        styles.outerBoxBase,
+                        hovered && styles.outerBoxHover,
+                        pressed && styles.outerBoxPressed,
+                    ]}
+                >
+                    <View
+                        style={[
+                            styles.base,
+                            tintColor != null
+                                ? {background: tintColor}
+                                : undefined,
+                            hovered && styles.hovered,
+                            focused && styles.focused,
+                            pressed && styles.pressed,
+                        ]}
+                    >
+                        {children}
+                    </View>
+                </View>
             </View>
         );
     }
 }
+
+function mapStateToProps(state: State) {
+    return {
+        gestureManager: state.gestures.gestureManager,
```

Comment:
The more I see this the more I think that if anything, this Gesture Manager should go in React context.